OpenRA.Mods.Common/AI/States/GroundStates.cs

```diff
@@ -140,9 +147,11 @@ public void Tick(Squad owner)
 				}
 			}
 
+			// Switch target during fight
```

Comment:
Not sure if I understood this here correctly, but aren't you trying to ensure that the AI is *not* switching targets during the fight?